kratos/utilities/force_and_torque_utils.cpp

```diff
@@ -0,0 +1,75 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+
+/* System includes */
+
+/* External includes */
+
+/* Project includes */
+#include "utilities/force_and_torque_utils.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/reduction_utilities.h"
+#include "utilities/math_utils.h"
+
+namespace Kratos
+{
+
+
+std::array<array_1d<double,3>,2> ForceAndTorqueUtils::SumForceAndTorque(const ModelPart& rModelPart,
+                                                                        const array_1d<double,3>& rReferencePoint)
+{
+    KRATOS_TRY
+
+    using CompoundArray = array_1d<double,6>;
+
+    CompoundArray force_and_moment = block_for_each<SumReduction<CompoundArray>>(
+        rModelPart.GetCommunicator().LocalMesh().Nodes(),
+        [&rReferencePoint](const Node<3>& rNode) -> CompoundArray
+        {
+            array_1d<double,3> force, moment;
+
+            force = rNode.GetSolutionStepValue(REACTION);
+
+            if (rNode.SolutionStepsDataHas(MOMENT)) {
```

Comment:
This check should be done once. Doing it outside the loop for the `begin` iterator is sufficient as the nodal based is the same.

Reply:
Will all nodes have ```MOMENT```s if the model consists of mixed elements?

Reply:
If the model consist of mixed elements all the elements would have the same DOFs (u-p, u-eps, u-eps_vol, v-p or whathever). If there are different element types in the mesh, DOFs would be different. If we want to consider this case, which es fair, I'd make the check optional.

In any case, note that after the creation of the "only-force" method the check would be no longer needed.